src/log_impl.rs

```diff
@@ -428,6 +446,12 @@ impl<'a> FormatCallback<'a> {
 // No need to write this twice (used for Stdout and Stderr structs)
 macro_rules! std_log_impl {
     ($ident:ident) => {
+        impl $ident {
+            pub fn new(stream: io::$ident) -> Self {
```

Comment:
I think `Stdout` and `Stderr` constructors, if public, should take no arguments, as the streams they wrap are singletons.